### PR TITLE
Ymd09 firmware update. Moved to data-driven, rgblight changed to rgb matrix.

### PR DESCRIPTION
**Pull request by SHVD3x**
## Description

Moved a lot of configuration setings to data driven config. Changed rgblight to rgb matrix for more effects. This change need VIA json config to be updated. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ymdk/ymd09/info.json [hunk 1]
```diff
@@ -10,19 +10,55 @@
...
+    "features": {
+        "bootmagic": true,
+        "mousekey": true,
+        "extrakey": true,
+        "console": false,
+        "command": false,
+        "nkro": true,
+        "rgb_matrix": true,
+        "lto": true,
+    },
```
Comment (waffle87) [inline keyboards/ymdk/ymd09/info.json:22]:
```suggestion
    },
    "build": {
        "lto": true
    },
```

File: keyboards/ymdk/ymd09/info.json [hunk 2]
```diff
@@ -10,19 +10,55 @@
...
+    "diode_direction": "ROW2COL",
+    "matrix_pins": {
+        "cols": ["D6", "D2", "D1"],
+        "rows": ["F5", "F4", "F1"]
+    },
+    "bootmagic": {
+        "matrix": [0, 1] // Hold 8 on start up to clear EEPROM and boot into bootloader mode
+    },
```
Comment (waffle87) [inline keyboards/ymdk/ymd09/info.json:30]: Why this change? That wasn't the prior behaviour nor should it be changed to something non-standard.

File: keyboards/ymdk/ymd09/info.json [hunk 3]
```diff
@@ -10,19 +10,55 @@
...
+                {"label": "6", "matrix": [1, 2], "x": 2, "y": 1},
+                {"label": "1", "matrix": [2, 0], "x": 0, "y": 2},
+                {"label": "2", "matrix": [2, 1], "x": 1, "y": 2},
+                {"label": "3", "matrix": [2, 2], "x": 2, "y": 2}
             ]
         }
+    },
+    "rgblight": {
+        "pin": "E2"
+    },
```
Comment (waffle87) [inline keyboards/ymdk/ymd09/info.json:48]:
```suggestion
```

File: keyboards/ymdk/ymd09/config.h [hunk 4]
```diff
@@ -17,33 +17,58 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #pragma once
 
+#define RGB_MATRIX_LED_COUNT 9
```
Comment (waffle87) [inline keyboards/ymdk/ymd09/config.h:20]:
```suggestion
#define RGB_MATRIX_LED_COUNT 9
#define RGB_DI_PIN E2
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/ymdk/ymd09/info.json (hunk 2 above)
Comment (SHVD3x) [reply keyboards/ymdk/ymd09/info.json:30]: Because this macropad does not have default bootmagic key at all. I can change readme file according to this change. The key can be changed to [0,0] to be more standart-like.

Comment (zvecr) [reply keyboards/ymdk/ymd09/info.json:32]: > The key can be changed to [0,0] to be more standart-like.

This should be done, random configuration just makes it more complicated for end users.

The default is `0,0`, so adding that is pointless. It should just rely on the defaults, and the current config removed.

File: keyboards/ymdk/ymd09/keymaps/default/keymap.c [hunk 5]
```diff
@@ -1,3 +1,18 @@
+/*
+Copyright 2023 SHVD3x
```
Comment (zvecr) [inline keyboards/ymdk/ymd09/keymaps/default/keymap.c:2]: You cant really claim copyright ownership on a file you have made zero changes to.

Comment (SHVD3x) [reply keyboards/ymdk/ymd09/keymaps/default/keymap.c:2]: This file didnt have copyright label at all. I dont know what should I write here, this is needed for lint check.

File: keyboards/ymdk/ymd09/info.json [hunk 6]
```diff
@@ -10,19 +10,51 @@
...
+            { "flags": 4, "matrix": [0, 1], "x": 12, "y": 0 },  //8
+            { "flags": 4, "matrix": [0, 0], "x":  0, "y": 0 },  //9
+            { "flags": 4, "matrix": [1, 2], "x": 24, "y": 12 }, //4
+            { "flags": 4, "matrix": [1, 1], "x": 12, "y": 12 }, //5
+            { "flags": 4, "matrix": [1, 0], "x":  0, "y": 12 }, //6
+            { "flags": 4, "matrix": [2, 2], "x": 24, "y": 24 }, //1
+            { "flags": 4, "matrix": [2, 1], "x": 12, "y": 24 }, //2
+            { "flags": 4, "matrix": [2, 0], "x":  0, "y": 24 }  //3
+        ],
+        "center_point": [ 12, 12 ],
```
Comment (zvecr) [inline keyboards/ymdk/ymd09/info.json:58]:
```suggestion
    "rgb_matrix": {
        "driver": "WS2812",
        "layout": [
            { "flags": 4, "matrix": [0, 2], "x": 64, "y": 0 },  //7
            { "flags": 4, "matrix": [0, 1], "x": 32, "y": 0 },  //8
            { "flags": 4, "matrix": [0, 0], "x":  0, "y": 0 },  //9
            { "flags": 4, "matrix": [1, 2], "x": 64, "y": 32 }, //4
            { "flags": 4, "matrix": [1, 1], "x": 32, "y": 32 }, //5
            { "flags": 4, "matrix": [1, 0], "x":  0, "y": 32 }, //6
            { "flags": 4, "matrix": [2, 2], "x": 64, "y": 64 }, //1
            { "flags": 4, "matrix": [2, 1], "x": 32, "y": 64 }, //2
            { "flags": 4, "matrix": [2, 0], "x":  0, "y": 64 }  //3
        ],
        "center_point": [ 32, 32 ],
```

**Review by waffle87 — APPROVED** [review]

File: keyboards/ymdk/ymd09/config.h [hunk 7]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2020 Patrick Fruh
+Copyright 2023 SHVD3x
```
Comment (drashna) [inline keyboards/ymdk/ymd09/config.h:2]: Original copyright shouldn't be removed. 

```suggestion
Copyright 2020 Patrick Fruh
Copyright 2023 SHVD3x
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]
